speck.c

```diff
@@ -6,6 +6,8 @@
 #include <stdlib.h>
 #include <stdint.h>
 
+#ifndef __ANDROID_NDK__
```

Comment:
Please move this android logic on a separate file. You can create empty hooks for platforms that do not need them, but I would like to see clean code. You can rework n2n in C++ if ti can help.

Reply:
Please see https://github.com/ntop/n2n/pull/259#discussion_r439337180 . Based on your comments it seems like the whole n2n codebase needs a rework, why do you want the rework to be part of this PR? Can this be just merged so that the android support is fixed and further work discussed separately?